File: concrete/src/main/java/pcb/uwu/unit/base/MinuteUnit.java
package pcb.uwu.unit.base;

import java.math.BigDecimal;

public class MinuteUnit extends TimeUnit {

	private static final BigDecimal SECONDS_PER_MINUTE = new BigDecimal(60);

	public static final MinuteUnit MINUTE = new MinuteUnit();

	private MinuteUnit() {
		super("min", "minute", "minutes",
				value -> value.times(SECONDS_PER_MINUTE),
				value -> value.div(SECONDS_PER_MINUTE));
	}
}

File: concrete/src/main/java/pcb/uwu/unit/base/MinuteUnit.kt
package pcb.uwu.unit.base

import java.math.BigDecimal

object MinuteUnit
    : TimeUnit(symbol = "min",
               singularName = "minute",
               pluralName = "minutes",
               translationToCanonical = { it * BigDecimal(60) },
               translationFromCanonical = { it / BigDecimal(60) })
